python/lsst/pipe/tasks/htmIndexEngine.py

```diff
@@ -133,6 +133,135 @@ def run(self, parsedCmd):
                 result = result,
             )
 
+class IngestIndexedReferenceTask(pipeBase.CmdLineTask):
+    ConfigClass = IngestIndexedReferenceConfig
+    RunnerClass = IngestReferenceRunner
+    _DefaultName = 'IngestIndexedReferenceTask'
+
+    _flags = ['photometric', 'resolved', 'variable']
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = pipeBase.ArgumentParser(name=cls._DefaultName)
+        parser.add_argument("files", nargs="+", help="Names of files to index")
+        return parser
+
+    def __init__(self, *args, **kwargs):
+        self.butler = kwargs.pop('butler')
+        pipeBase.Task.__init__(self, *args, **kwargs)
+        self.indexer = HtmIndexer(self.config.level)
+
+    def create_indexed_catalog(self, files):
+        rec_num = 0
+        first = True
+        for filename in files:
+            names = True
+            if self.config.colnames:
+                names = self.config.colnames
+            arr = numpy.genfromtxt(filename, dtype=None, skip_header=self.config.header_lines,
```

Comment:
Text input is the less trivial of the two.  I figured a FITS reader could be added once this is in.  I could do it here if it's a no-go without it.